File: biddo_client/lib/core/models/currency.dart
import '../../utils/constants.dart';

class Currency {
  String id;
  String code;
  String symbol;
  Map<String, String> name;

  Currency({
    required this.id,
    required this.code,
    required this.name,
    required this.symbol,
  });

  static Currency fromJSON(dynamic data) {
    return Currency(
      id: data['id'],
      code: data['code'],
      name: Map.fromEntries(
        Constants.LANGUAGES.map((lang) => MapEntry(
              lang['code']!,
              data['name'][lang['code']] ?? '',
            )),
      ),
      symbol: data['symbol'],
    );
  }
}

File: core/domain/currency.ts
export class Currency {
  id: string
  code: string
  symbol: string
  name: Record<string, string>

  constructor(params: { id: string; code: string; symbol: string; name: Record<string, string> }) {
    this.id = params.id
    this.code = params.code
    this.symbol = params.symbol
    this.name = params.name
  }

  static fromJSON(data: Record<string, unknown> = {}): Currency {
    return new Currency({
      id: data.id as string,
      code: data.code as string,
      symbol: data.symbol as string,
      name: data.name as Record<string, string>,
    })
  }
}
